src/Sulu/Bundle/TestBundle/Testing/KernelTestCase.php

```diff
@@ -0,0 +1,190 @@
+ */
+
+namespace Sulu\Bundle\TestBundle\Testing;
+
+use Doctrine\Common\DataFixtures\Executor\ORMExecutor;
+use Doctrine\Common\DataFixtures\ProxyReferenceRepository;
+use Doctrine\Common\DataFixtures\Purger\ORMPurger;
+use Doctrine\ORM\EntityManager;
+use InvalidArgumentException;
+use Sulu\Bundle\ContentBundle\Document\HomeDocument;
+use Sulu\Bundle\TestBundle\Kernel\SuluTestKernel;
+use Sulu\Component\Content\Document\WorkflowStage;
+use Symfony\Bundle\FrameworkBundle\Client;
+use Symfony\Component\Security\Core\Tests\Authentication\Token\TestUser;
+
+/**
+ * Base test case for functional tests in Sulu.
+ */
+abstract class KernelTestCase extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var Container
+     */
+    private $container;
+
+    /**
+     * @var KernelInterface[]
+     */
+    private $kernelStack;
+
+    /**
+     * Return the test container.
+     *
+     * This container will use the default configuration of the kernel.
+     *
+     * If you require the container with a different kernel configuration
+     * you should create a new Kernel with the `getKernel` method and
+     * retrieve the Container from that.
+     *
+     * @return ContainerInterface
+     */
+    public function getContainer()
+    {
+        if ($this->container) {
+            return $this->container;
+        }
+
+        $this->container = $this->getKernel($this->getKernelConfiguration())->getContainer();
+
+        return $this->container;
+    }
+
+    /**
+     * Create and return new test kernel and pass the sulu.context to it.
+     *
+     * All kernels created will be added to the kernel stack, at the end of
+     * each test tearDown should invoke the $this->shutdownKernels() which
+     * will shutdown each of the kernels and remove each from the stack.
+     *
+     * The kernel which has been "found" MUST implement the SuluTestCase.
+     *
+     * @throws InvalidArgumentException
+     */
+    protected function getKernel(array $options = [])
+    {
+        $this->requireKernel();
+
+        $options = array_merge([
+            'environment' => 'test',
+            'debug' => true,
+            'sulu_context' => 'admin'
+        ], $this->getKernelConfiguration(), $options);
+
+        $kernel = new \AppKernel(
+            $options['environment'],
+            $options['debug'],
+            $options['sulu_context']
+        );
+
+        if (!$kernel instanceof SuluTestKernel) {
+            throw new \InvalidArgumentException(sprintf(
+                'All Sulu testing Kernel classes must extend SuluTestKernel, "%s" does not',
+                get_class($kernel)
+            ));
+        }
+
+        $kernel->boot();
+        $this->kernelStack[] = $kernel;
+
+        return $kernel;
+    }
+
+    /**
+     * Return the kernel confguration for the base test case.
+     *
+     * Typically you will not need to override this (a notable exception is if
+     * you need to use the "website" context).
```

Comment:
This is an anti-pattern, maybe advise against its use. In cases of the "website" context, you can equally use the getKernel method.